test/test_torch.py

```diff
@@ -9040,6 +9040,18 @@ def test_kthvalue(self, device, dtype):
             self.assertEqual(res1val[:, :], res2val[:, :, k - 1], atol=0, rtol=0)
             self.assertEqual(res1ind[:, :], res2ind[:, :, k - 1], atol=0, rtol=0)
 
+    # test overlapping output
+    @dtypes(torch.double)
+    @onlyOnCPUAndCUDA   # Fails on XLA
+    def test_kthvalue_overlap(self, device, dtype):
+        S = 10
+        k = 5
+        a = torch.randn(S)
+        values = torch.empty_like(a)
```

Comment:
you don't need `values`

Reply:
Done! 
Besides, due to the rebase conflicts, I merge the last  serveral commits to only one and forced a push to branch.
Thanks for the review!